features/behat-steps.feature

```diff
@@ -0,0 +1,741 @@
+      """
+    Then the wp-cli.yml file should exist
+    And the mock-requests.php file should exist
+
+  @require-wp
+  Scenario: Test background process launch
+    Given a WP installation
+    When I launch in the background `wp eval 'sleep(2); echo "done";'`
+    # Background process should not block, continuing immediately
+
+  @require-wp
+  Scenario: Test custom wp-content directory
+    Given a WP installation
+    And a custom wp-content directory
+    Then the my-content directory should exist
+    And the my-plugins directory should exist
+    And the my-mu-plugins directory should exist
+    And the wp-config.php file should contain:
+      """
+      WP_CONTENT_DIR
+      """
+
+  @require-wp
+  Scenario: Test WP multisite subdirectory installation
+    Given a WP multisite subdirectory installation
+    When I run `wp site list --field=url`
+    Then STDOUT should not be empty
+    And the return code should be 0
+
+  @require-wp
+  Scenario: Test WP multisite subdomain installation
+    Given a WP multisite subdomain installation
+    When I run `wp site list --field=url`
+    Then STDOUT should not be empty
+    And the return code should be 0
+
+  @require-wp
+  Scenario: Test misconfigured WP_CONTENT_DIR
+    Given a WP installation
+    And a misconfigured WP_CONTENT_DIR constant directory
+    Then the wp-config.php file should contain:
+      """
+      define( 'WP_CONTENT_DIR', '' );
+      """
+
+  @require-download
+  Scenario: Test download step
+    Given an empty cache
+    And download:
+      | path                       | url                                    |
+      | {SUITE_CACHE_DIR}/test.txt | https://www.iana.org/robots.txt        |
+    Then the {SUITE_CACHE_DIR}/test.txt file should exist
+
+  @require-wp @require-composer
+  Scenario: Test WP installation with Composer
+    Given a WP installation with Composer
+    Then the composer.json file should exist
+    And the vendor directory should exist
+    When I run `wp core version`
+    Then STDOUT should not be empty
+
+  @require-wp @require-composer
+  Scenario: Test WP installation with Composer and custom vendor directory
+    Given a WP installation with Composer and a custom vendor directory 'custom-vendor'
+    Then the composer.json file should exist
+    And the custom-vendor directory should exist
+
+  @require-wp @require-composer
+  Scenario: Test dependency on current wp-cli
+    Given a WP installation with Composer
+    And a dependency on current wp-cli
+    Then the composer.json file should contain:
+      """
+      wp-cli/wp-cli
+      """
+
+  @require-wp @require-php-server
+  Scenario: Test PHP built-in web server
+    Given a WP installation
+    And a PHP built-in web server
+    Then the HTTP status code should be 200
+
+  @require-wp @require-php-server
+  Scenario: Test PHP built-in web server with subdirectory
+    Given an empty directory
+    And an empty wordpress directory
+    And a WP installation in 'wordpress'
```

Comment:
This currently triggers a PHP warning:

```
Warning: mkdir(): File exists in src/Context/FeatureContext.php line 1306
```

Need to remove the `And an empty wordpress directory` step above.